molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/entity/impl/SnpEffAnnotator.java

```diff
@@ -72,102 +51,46 @@
 	private static final Logger LOG = LoggerFactory.getLogger(SnpEffAnnotator.class);
 	public static final String NAME = "snpEff";
 
-	public static final String ANNOTATION = "Annotation";
-	public static final String PUTATIVE_IMPACT = "Putative_impact";
-	public static final String GENE_NAME = "Gene_Name";
-	public static final String GENE_ID = "Gene_ID";
-	public static final String FEATURE_TYPE = "Feature_type";
-	public static final String FEATURE_ID = "Feature_ID";
-	public static final String TRANSCRIPT_BIOTYPE = "Transcript_biotype";
-	public static final String RANK_TOTAL = "Rank_total";
-	public static final String HGVS_C = "HGVS_c";
-	public static final String HGVS_P = "HGVS_p";
-	public static final String C_DNA_POSITION = "cDNA_position";
-	public static final String CDS_POSITION = "CDS_position";
-	public static final String PROTEIN_POSITION = "Protein_position";
-	public static final String DISTANCE_TO_FEATURE = "Distance_to_feature";
-	public static final String ERRORS = "Errors";
-	public static final String LOF = "LOF";
-	public static final String NMD = "NMD";
-
 	public enum Impact
 	{
 		MODIFIER, LOW, MODERATE, HIGH
 	}
 
 	@Autowired
-	private JarRunner jarRunner;
+	private SnpEffRunner snpEffRunner;
 
 	@Autowired
 	private Entity snpEffAnnotatorSettings;
 
-	@Bean
-	public RepositoryAnnotator snpEff()
-	{
-		return new SnpEffRepositoryAnnotator(snpEffAnnotatorSettings, jarRunner);
-	}
+	@Autowired
+	private DataServiceImpl dataService;
 
 	@Bean
-	JarRunner jarRunner()
-	{
-		return new JarRunnerImpl();
-	}
-
-	/**
-	 * Helper function to get gene name from entity
-	 *
-	 * @param entity
-	 * @return
-	 */
-	public static String getGeneNameFromEntity(Entity entity)
+	public RepositoryAnnotator snpEff()
 	{
-		String geneSymbol = null;
-		if (entity.getString(GENE_NAME) != null)
-		{
-			geneSymbol = entity.getString(GENE_NAME);
-		}
-		if (geneSymbol == null)
-		{
-			String annField = entity.getString("ANN");
-			if (annField != null)
-			{
-				// if the entity is annotated with the snpEff annotator the split is already done
-				String[] split = annField.split("\\|", -1);
-				// TODO: ask Joeri to explain this line
-				if (split.length > 10)
-				{
-					// 3 is 'gene name'
-					// TODO check if it should not be index 4 -> 'gene id'
-					if (split[3].length() != 0)
-					{
-						geneSymbol = split[3];
-					}
-					else
-					{
-						// will happen a lot for whole genome sequencing data
-						LOG.info("No gene symbol in ANN field for " + entity.toString());
-					}
-
-				}
-			}
-		}
-		return geneSymbol;
+		return new SnpEffRepositoryAnnotator(snpEffRunner, snpEffAnnotatorSettings, dataService);
 	}
 
-	public static class SnpEffRepositoryAnnotator extends AbstractRepositoryAnnotator
+	public static class SnpEffRepositoryAnnotator extends AbstractRepositoryAnnotator implements RefEntityAnnotator
```

Comment:
Make a package that contains all SnpEff stuff.
1. And Create SnpEffAnnotatorConfig
2. SnpEffRepositoryAnnotator.java

For searchability in Eclipse
Name conventions? When @Configuration then Config at the end